spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClientBuilder.java

```diff
@@ -62,27 +54,40 @@ public CommitOwnerClient build(SparkSession spark, Map<String, String> conf) {
         String dynamoDBEndpoint = conf.get(DYNAMO_DB_ENDPOINT_KEY).getOrElse(() -> {
             throw new RuntimeException(DYNAMO_DB_ENDPOINT_KEY + " not found");
         });
+        String awsCredentialsProviderName =
+                spark.conf().get(DeltaSQLConf.MANAGED_COMMIT_DDB_AWS_CREDENTIALS_PROVIDER_NAME());
+        int readCapacityUnits = Integer.parseInt(
+                spark.conf().get(DeltaSQLConf.MANAGED_COMMIT_DDB_READ_CAPACITY_UNITS().key()));
+        int writeCapacityUnits = Integer.parseInt(
+                spark.conf().get(DeltaSQLConf.MANAGED_COMMIT_DDB_WRITE_CAPACITY_UNITS().key()));
+        boolean skipPathCheck = Boolean.parseBoolean(
+                spark.conf().get(DeltaSQLConf.MANAGED_COMMIT_DDB_SKIP_PATH_CHECK().key()));
         try {
-            AmazonDynamoDBClient client =
-                    createAmazonDDBClient(dynamoDBEndpoint, AWS_CREDENTIALS_PROVIDER);
+            AmazonDynamoDBClient ddbClient = createAmazonDDBClient(
+                    dynamoDBEndpoint,
+                    awsCredentialsProviderName,
+                    spark.sessionState().newHadoopConf()
+            );
             return new DynamoDBCommitOwnerClient(
-                    managedCommitsTableName, dynamoDBEndpoint, client, BACKFILL_BATCH_SIZE);
+                    managedCommitsTableName,
+                    dynamoDBEndpoint,
+                    ddbClient,
+                    BACKFILL_BATCH_SIZE,
+                    readCapacityUnits,
+                    writeCapacityUnits,
+                    skipPathCheck);
         } catch (Exception e) {
             throw new RuntimeException("Failed to create DynamoDB client", e);
         }
     }
 
     private AmazonDynamoDBClient createAmazonDDBClient(
             String endpoint,
-            String credentialProviderName
-    ) throws NoSuchMethodException,
-            ClassNotFoundException,
-            InvocationTargetException,
-            InstantiationException,
-            IllegalAccessException {
-        Class<?> awsCredentialsProviderClass = Class.forName(credentialProviderName);
+            String credentialProviderName,
+            Configuration hadoopConf
+    ) throws ReflectiveOperationException {
         AWSCredentialsProvider awsCredentialsProvider =
-                (AWSCredentialsProvider) awsCredentialsProviderClass.getConstructor().newInstance();
+                ReflectionUtils.createAwsCredentialsProvider(credentialProviderName, hadoopConf);
```

Comment:
Is this similar to how DynamoDBLogStore do it?

Reply:
Yes, this has been directly copied from there.